plugin-api/src/main/java/tech/pegasys/pantheon/plugin/services/StorageService.java

```diff
@@ -35,4 +36,13 @@
    * @return full set of possible segments required from the storage service.
    */
   List<SegmentIdentifier> getAllSegmentIdentifiers();
+
+  /**
+   * Retrieves a registered factory corresponding to the supplied factory name
+   *
+   * @param name The name of the factory to retrieve
+   * @return an optional containing the instance of the registered factory, or empty if the factory
+   *     hasn't been registered.
+   */
+  Optional<KeyValueStorageFactory> getByName(final String name);
```

Comment:
`final` modifier not needed on an interface

Reply:
nice find. done